config/01-ce-router-defaults.conf

```diff
@@ -151,89 +150,83 @@ JOB_ROUTER_TRANSFORM_OrigRequests @=jrt
         # HTCondor slot
         SET GlideinCpusIsGood      int(MATCH_EXP_JOB_GLIDEIN_Cpus ?: "0") isnt error
         # Also used by JobGpus and JobMemory
-        SET JobIsRunning           (JobStatus =!= 1) && (JobStatus =!= 5) && GlideinCpusIsGood
+        SET JobIsRunning           $(MY.JobIsRunning) && GlideinCpusIsGood
+        COPY RequestCpus           OriginalCpus
         SET JobCpus                JobIsRunning ? int(MATCH_EXP_JOB_GLIDEIN_Cpus) : OriginalCpus
         SET RequestCpus            TARGET.TotalCpus ?: JobCpus
     endif
 
-    EVALMACRO test_xcount $(MY.xcount:0) > 1
-    if $(test_xcount)
-        cpus = $(MY.xcount)
-    else
-        EVALMACRO test_orig_RequestCpus $(MY.orig_RequestCpus:0) > 1
-        if $(test_orig_RequestCpus)
-            cpus = $(MY.orig_RequestCpus)
-         else
-            cpus = $(default_xcount:1)
-         endif
-    endif
-
-    EVALSET OriginalCpus           $(cpus)
-    EVALSET remote_SMPGranularity  $(cpus)
-    EVALSET remote_NodeNumber      $(cpus)
 
-    DEFAULT JobIsRunning           (JobStatus =!= 1) && (JobStatus =!= 5)
-    DEFAULT JobCpus                OriginalCpus
-    DEFAULT RequestCpus            OriginalCpus
 @jrt
 
 
 JOB_ROUTER_TRANSFORM_Gpus @=jrt
     # Request GPUs for whole node jobs (HTCONDOR-103)
     # If a whole node job requests GPUs and is matched to a machine with GPUs then set the job's RequestGPUs to all the
     # GPUs on that machine
+    REQUIREMENTS RequestGpus isnt undefined
     if $(test_want_whole_node)
-        SET JOB_GLIDEIN_GPUs       "$$(MY.TotalGPUs ?: JobGPUs)"
+        SET JOB_GLIDEIN_GPUs       "$$([MY.TotalGPUs ?: JobGPUs])"
         # MATCH_EXP_JOB_GLIDEIN_GPUs is based on the value of JOB_GLIDEIN_GPUs once the routed job is matched to an
         # HTCondor slot
         SET GlideinGPUsIsGood      int(MATCH_EXP_JOB_GLIDEIN_GPUs ?: "0") isnt error
+        COPY RequestGPUs           OriginalGPUs
         SET JobGPUs                JobIsRunning ? int(MATCH_EXP_JOB_GLIDEIN_GPUs) : OriginalGPUs
         SET RequestGPUs            (TARGET.TotalGPUs > 0) ? TotalGPUs: JobGPUs
     endif
-
-    EVALSET OriginalGPUs           $(MY.orig_RequestGPUs)
-    DEFAULT RequestGPUs            OriginalGPUs
 @jrt
 
 
 JOB_ROUTER_TRANSFORM_Memory @=jrt
-    # Outside of the HTCondor WholeNodeJobs case, set RequestCpus to one of the following, in order:
-    # 1. 'maxMemory' from the source job if it's positive
-    # 2. RequestMemory from the source job
-    # 3. default_MaxMemory from the job route
-    # 4. 2000
+    # Outside of the HTCondor WholeNodeJobs case, set RequestMemory to one of the following, in order:
+    # 1. 'maxMemory' from the source job if it's defined
+    # 2. default_MaxMemory from the job route
+    # 3. 2000
+    # Note that if the route does the condor thing and sets RequestMemory, this transform will stomp on that
+
+    EVALSET RequestMemory maxMemory ?: ($(default_maxMemory:2000))
```

Comment:
For my own edification, why do we need the outer parens around `$default_maxMemory:2000`?

Reply:
default_maxMemory can be an expression here